packages/api/endpoints/granules.js

```diff
@@ -908,6 +910,36 @@ async function getByGranuleId(req, res) {
   return res.send({ ...result, recoveryStatus });
 }
 
+/**
+ * Based on a move-collections-task, will update a list of
+ * moved granules' records in PG and ES
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function updateGranulesAndFilesCollectionRecords(req, res) {
+  const {
+    knex = await getKnexClient(),
+    esClient = await getEsClient(),
+  } = req.testContext || {};
+  const payload = req.body;
+  try {
+    await updateGranulesAndFiles(knex, payload);
```

Comment:
🤔  

tl;dr. 

We should probably move the concurrency logic to the API package, handle bounding/configuring concurrency and address validation via something like zod. 

Details/concerns: 

1) We should consider implementing a concurrency or size bound somehow  the API is limited by design to 2 DB connections >by design<, if you throw a large enough request at this endpoint I'd expect it will either timeout or overwhelm the knex connection pool, or both.    That's *probably* not appropriate to the db package (and why we wound up with this sort of logic in the `api/lib` write 'stuff' as an approach. 

2) Users aren't going to just use this for the intended approach for 3757, if it's part of the public API we likely should use zod to validate inputs/schemas/etc before passing a payload to a @cumulus/db method.

Reply:
I don't think we want to make a bulk update feature like this utilize a single granule call - that's going to force scaling of lambda invocations *per granule*, and is super inefficient.       

There's room for various approaches here, but particularly, we need to put configurations/limits on concurrency around `await Promise.all(granules.map(async (granule) => {` so that we don't overwhelm the limited knex connection pool *or* make an exception for this bulk call and increase it, however that has user monitoring/tuning concerns. 

As an opinionated take:  I'd expect that best looks like having the DB package implement a single granule + all files update and associated tests, with an api lib method that does it for a bunch of granules and has concurrency controls in it.    That allows for solid testing around a single granule update,and decouples concurrency concerns from the DB package.        If the team feels strongly we should support bulk updates as part of the domain of the DB package, we could easily do that same approach entirely within the DB package.